### PR TITLE
Adds pronouns

### PR DESCRIPTION
**Pull request by vitorpamplona**
🤡

### CODE REVIEW

**Comment by rabble** [comment]
Looks good to me. It's entirely optional and users can put what ever text they want. Super flexible able to handle they/them, ze/zer, and Elon Musk's confusing prefered pronouns of Prosecute/Fauci.

**Comment by barrydeen** [comment]
🤡🤡🤡🤡

**Comment by vitorpamplona** [comment]
Nice! I am happy @rabble cares enough to move this forward for real. Let's go! 🚀

Ps.. I am going to start sending all my or as jokes since the rate of approval seems to be better. 😅

**Comment by laanwj** [comment]
> Super flexible able to handle they/them, ze/zer, and Elon Musk's confusing prefered pronouns of Prosecute/Fauci.

yes... perfect

**Comment by melvincarvalho** [comment]
@vitorpamplona, using a clown icon to mock people can come across as unprofessional. Nostr is a protocol, not a cult, and it's important that we avoid reinforcing any negative far right reputations or misunderstandings around it.

I wasn't aware that @rabble had merge access, but it’s good to see. 

User should control profiles not devs; in Mastodon, for instance, users can add four custom fields, while Solid allows unlimited customization.  We need this in nostr.

After five years, it would be great to expand Nostr profiles with features like [pubky](https://pubky.tech/) and other fields. I've been documenting profile field extensibility here for reference: 

https://w3id.org/nostr

**Comment by melvincarvalho** [comment]
Hi @Zk2u,

These kind of issues and comments pose a risk to the entire project, especially in the protocol area, where we are bound by GitHub's [Terms of Service](https://docs.github.com/en/site-policy/acceptable-use-policies/github-hate-speech-and-discrimination). Specifically:

- **Hostile Tone**: Statements like "You people need to go outside and touch some grass" and "Bring back bullying, jeez. Y’all are embarrassments" are dismissive and insulting. This type of language fosters a hostile environment, discourages collaboration, and is not aligned with GitHub’s requirement to maintain a welcoming and respectful space for contributors.

- **Mocking of Inclusivity**: The use of a clown emoji to mock the addition of pronouns undermines efforts to make the project more inclusive. Such behavior alienates contributors, particularly those who value inclusivity, and creates unnecessary division within the community.

As this project operates within GitHub's ecosystem (at least for now until we move to ngit), we have to adhere to its guidelines or it creates unnecessary problems for contributors and the project area.  

Consider a more constructive effort such as here:  https://github.com/nostr-protocol/nips/pull/1593

**Comment by fiatjaf** [comment]
Anyone can add this field if they want, but NIP-24 is meant to keep track of these extra fields that have already become de facto standards.

A more reasonable approach would be to open a serious pull request (one that isn't a ill-intentioned bad joke meant to trigger leftists or right-wingers) as a way to notify clients of the possibility, then if many of them like it and implement it we proceed to add the field to NIP-24.

**Comment by vitorpamplona** [comment]
Boo.. the dictator doesn't like fun when it is against his ideology.

**Comment by barrydeen** [comment]
The place to have fun is in the clients. I would suggest adding pronouns to amethyst to show your support instead of playing games in the protocol repo.

**Comment by melvincarvalho** [comment]
> revert improperly  

Would you consider letting the community decide on changes like this instead of one person taking control? I think a more open process would be better for Nostr and for you as well, helping the project grow stronger.  

> trigger leftists  

Could I also suggest avoiding terms like this in protocol discussions? People outside the project notice and comment on these things, which can hurt Nostr’s image. For example, on the [ActivityPub forum](https://socialhub.activitypub.rocks/t/is-this-the-trump-guy/4634/8), someone said:  

> Looking more broadly, Nostr’s a tricky situation in that their pseudonymous founder has “leftism is a virus” on his profile  

This might be unfair, but as a leader, your words set the tone. Keeping the protocol area neutral and focused on technology would help Nostr grow and attract more people.  Also pronouns, love them or hate them, are a de-facto standard across the entire open social web.  Hopefully we can get a good solution with Alex's compromise.  Thanks for considering!

**Comment by vitorpamplona** [comment]
> The place to have fun is in the clients. I would suggest adding pronouns to amethyst to show your support instead of playing games in the protocol repo.

I was waiting for resolution on #1593, but sure: https://github.com/vitorpamplona/amethyst/commit/4a80a16473f6a0357a367e40836089c16ff463f3

**Comment by alexgleason** [comment]
The rule we decided on was that 2+ implementations need to exist before merging. This is a reasonable criteria that I think everyone should agree with.

I think fiatjaf's response was balanced.

> trigger leftists or right-wingers

And then he left an open door:

> if many of them like it and implement it we proceed to add the field to NIP-24

It was actually Rabble who broke the rules. But only because he called Vitor's bluff. :smiley: It was good entertainment, but not actually something worth fighting about.

**Comment by AsaiToshiya** [comment]
> I think fiatjaf's response was balanced.

Agreed.

**Comment by SilberWitch** [comment]
This was legit just 
- a political stunt, for the lulz
- the change makes zero sense, so it wouldn't have gotten past a proper review (It's just an additional freetext field; unsortable, unfilterable, and with no functional purpose. How many memo fields does one profile need?) 
- it appears that two developers proposed, approved, and merged the commit. That's outrageous, when the rest of us get blocked by a small army of approvers, every darn time, especially on a change to something pivotal like Kind 0.

**Comment by rabble** [comment]
Here's the second implimentation of this change to the nip. 

https://github.com/planetary-social/nos/pull/1695

I think the custom fields option that @alexgleason proposed #1593 could work but i'd like to see something there which defines conventions so folks know that it's "pronouns" vs "Pronouns" vs 'pronoun'. Basically make it clear what the conventions are for suggesting commonly used labels.

**Comment by melvincarvalho** [comment]
> rprounouns cannot be removed
> 
> https://npronouns.github.io/
> 
> you believe in freedom and i will show what is freedom

Please do not use this, it overwrites all your profile data, so users are getting rekt

File: 24.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ These are extra fields not specified in NIP-01 that may be present in the string
   - `website`: a web URL related in any way to the event author.
   - `banner`: an URL to a wide (~1024x768) picture to be optionally displayed in the background of a profile screen.
   - `bot`: a boolean to clarify that the content is entirely or partially the result of automation, such as with chatbots or newsfeeds.
+  - `pronouns`: a string representing the preferred way to refer to this person.
```
Comment (RandyMcMillan) [inline 24.md:20]: 👀

**Comment by melvincarvalho** [comment]
> I think that's the point.

It's not a good idea, as it could make Nostr unusable for new users and even risk them losing funds—in order to make a political statement. Nostr is a protocol, not a cult.

**Comment by SilberWitch** [comment]
Yeah, it's not helpful. Guess he wanted to protest.

**Pull request merged**